omegaconf/omegaconf.py

```diff
@@ -898,6 +898,10 @@ def _node_wrap(
     ref_type: Any = Any,
 ) -> Node:
     node: Node
+    allow_objects = parent is not None and parent._get_flag("allow_objects") is True
+    flags = {"allow_objects": allow_objects} if allow_objects is not None else {}
+    dummy = OmegaConf.create(flags=flags)
```

Comment:
This is fixing the issue you pointed out. if you think there is another one please show it.

Reply:
> Can you demonstrate a minimal repro of the problem? It would be easier than me guessing that you are thinking about.

I just created a new issue for the bug with `flag_override`: #589 589

Reply:
> This is fixing the issue you pointed out. if you think there is another one please show it.

Which issue btw? (I don't actually remember pointing out #586)

Although your fix is indeed fixing #586, the same problem may occur in any other place where we create a `DictConfig` with a parent in struct mode. As far as I can tell by looking at the code there is no other place where this can happen in OmegaConf at this time, but it could happen in the future. So I'm suggesting to fix the root cause instead, which should fix both t#586 and any future occurrence of DictConfig creation with a parent in struct mode.

The fix I'm proposing is simply this in `DictConfig._set_value_impl()`:
```python
            elif isinstance(value, dict):
                with flag_override(self, "struct", False):
                    with flag_override(self, "readonly", False):
                        for k, v in value.items():
                            self.__setitem__(k, v)
```

(but it requires fixing #589 first as otherwise other stuff break)

Reply:
[This](https://github.com/omry/omegaconf/pull/581/files#r589151713) is where you reported it.

Reply:
Basically what's happening in `_node_wrap()`. I don't expect any user to do that (which is why I didn't show a repro), but it might happen in the future as OmegaConf's internals evolve.

Something like this:
```python
c = OmegaConf.create({"a": "???"})
OmegaConf.set_struct(c, True)
# `future_a` is a dictionary that we intend to be assigned to `c.a` later,
# and for some reason we need to have its parent set properly before that happens
future_a = DictConfig({"b": 0}, key="a", parent=c)  # <== crash here
```

Reply:
> > Which issue btw? (I don't actually remember pointing out #586)
> 
> [This](https://github.com/omry/omegaconf/pull/581/files#r589151713) is where you reported it.

Ah my bad, totally forgot about it, and missed that notification somehow. Thanks! :)

Reply:
> > Which issue btw? (I don't actually remember pointing out #586)
> 
> [This](https://github.com/omry/omegaconf/pull/581/files#r589151713) is where you reported it.

Ah my bad, totally forgot about it, and missed that notification somehow. Thanks! :)